### DISCUSSION TITLE
Talk about the main differences between netbeans and vscode?

### DISCUSSION THREAD

**Discussion by oceanfish567** [apache/netbeans#6924] [General]
What are the main differences between netbeans and vscode?
For the purpose of website development, using html/css/js/php software, can vscode replace netbeans? Or should we say that netbeans has feature and advantages that vscode cannot replace?

I have been using Netbeans and vscode for website development for more than 2 years. I really can’t tell the difference between the two softwares. Is it possible that I only use them for website development and production?
All my friends have switched to vscode. It is undeniable that vscode runs extremely fast, has many packages, and has a large community.
However, vscode cannot open multiple and multiple software projects like Netbeans. It can only open one project at a time. It is a nightmare for engineers who need to collaborate between different projects. Another nightmare is that vscode may be too free in nature. It is not easy and complicated to set up the environment, set up the package, set up debugging, etc.
I think the advantage of netbeans is that it can open multiple projects for collaboration at the same time, and various settings are easy. Disadvantages: The running speed is too slow, and the development of various packages is too slow or not developed. For example, Laravel, which I use, has a huge customer base, but netbeans does not support its blade, Eloquent ORM, etc. There are also no command line tools provided.
Even so, as long as vsocde still doesn't have the function of opening multiple projects, I'm sure the two softwares should be used together. What do you think?

**Comment by bobcbui** [comment]
For me, one project per window may be an advantage

**Comment by parallels999** [comment]
> but netbeans does not support its blade

https://github.com/haidubogdan/netbeans-php-blade-plugin (_It works_)
https://github.com/haidubogdan/nb-18-php-blade-plugin/tree/main/php-laravel (_Still under dev_)

https://github.com/apache/netbeans/issues/5965#issuecomment-2030647953

**Comment by fredsal** [comment]
- https://github.com/apache/netbeans/discussions/7231

**Comment by pincher2012** [comment]
Netbeans has native support of refactoring features, like renaming. Renaming in VSCode is paid feature of some extensions.

**Reply by Chris2011** [reply]
for example? In plain VS Code it is working as well. We are not talking about extensions.

**Reply by pincher2012** [reply]
> For the purpose of website development, using html/css/js/php software, can vscode replace netbeans?

Vscode does not have native support of PHP

**Comment by Chris2011** [comment]
@oceanfish567 
Here is my oppinion:

> For the purpose of website development, using html/css/js/php software, can vscode replace netbeans? Or should we say that netbeans has feature and advantages that vscode cannot replace?

Yes definitely could VS Code replace NetBeans but first nothing should replace the other one. It is about competitors everytime w/o having a monopole. Google is owning the search, Open AI ChatGPT, MS their operating systems and for IDE VS Code or JetBrains. Everyone has their advantages and disadvantages.

The problem here is that the community, in my perspective, for NetBeans is not thaaaat high and NetBeans also lacks behind a lot of features. I see it everyday. And unfortunately, I use VS Code for my daily work because first: It has quality support for Angular and Vue. For my daily work, And second I don't want to create my working tools on demand when I need it. I'm just a user for this. And yes I use the most of the features of VS Code which they have, for the mentioned support.

Also the problem of adding more support of stuff that come "daily" for example JS Frameworks and and and is, that NetBeans is historically grown, it has a huge code base with legacy stuff which was never made for modern stuff. We see it right now, that we have problems of adding new stuff to it. I also see it when I try to stuff to it. Anyway but, NetBeans history was for Java and later PHP and C/C++, Later git and HTML came but quite late and just basics. All of us needs to make this better if we want to. But some people like me have not enough time and basically for me, not the best knowledge of the deep deep integration for bigger addings/changes, etc. I do my best what I can but time is limited. My wish is to own a company and have people, developing the stuff for NetBeans including me but who wants to pay us?

> However, vscode cannot open multiple and multiple software projects like Netbeans. It can only open one project at a time. It is a nightmare for engineers who need to collaborate between different projects. 

I will be honest and maybe you will leave NetBeans then but this is not a secret, of course you can add more than one project to the workspace. Just right click on the projects tab and choose add folder to workspace. VS Code will recognice everything inside of it also that it is a git project.

And yes I'm using both. NetBeans more for NetBeans stuff and sometimes for simple Web Projects without any framework and PHP and VS Code for the rest. Let's choose the right tool for your work.
